crates/core/database/src/models/users/model.rs

```diff
@@ -297,23 +302,28 @@ impl User {
             }
         }
 
-        // Ensure none of the following substrings show up in the username
-        const BLOCKED_SUBSTRINGS: &[&str] = &[
-            "```",
-            "discord.gg",
-            "rvlt.gg",
-            "guilded.gg",
-            "https://",
-            "http://",
-        ];
-
-        for substr in BLOCKED_SUBSTRINGS {
-            if username_lowercase.contains(substr) {
-                return Err(create_error!(InvalidUsername));
-            }
+        Ok(())
+    }
+
+    /// sanitise username
+    fn sanitise_username(username: &str) -> (bool, String) {
+        let original_username = username;
+        // sanitise homoglyphs
+        let options = decancer::Options::default()
+            .retain_capitalization();
+        let mut username = decancer::cure(&username, options).unwrap().to_string();
+
+        if BLOCKED_USERNAME_PATTERNS.is_match(&username) {
+            username = BLOCKED_USERNAME_PATTERNS.replace_all(&mut username, "").into_owned();
         }
 
-        Ok(username)
+        const USERNAME_MIN_LEN: usize = 2;
+        let username_length_diff = USERNAME_MIN_LEN.saturating_sub(username.len());
+        if username_length_diff > 0 {
+            username.push_str(&"_".repeat(username_length_diff))
+        }
+
+        (original_username != username, username)
```

Comment:
returning a boolean here is strange considering its usages already hold the original name, and it can simply be checked out there.